src/Graph/DOT.elm

```diff
@@ -0,0 +1,172 @@
+string over a port to the javascript version of the GraphViz library,
+<https://github.com/mdaines/viz.js/> (see the examples there fore more
+specifics on how to embed the generated visualization).
+
+@docs output
+
+
+# Attrs
+
+GraphViz allows for customizing the graph's look via "Attrs."
+
+@docs Styles, Rankdir, defaultStyles, outputWithStyles, outputWithStylesAndAttributes
+
+-}
+
+import Dict exposing (Dict)
+import Graph exposing (Edge, Graph, Node, edges, get, nodes)
+
+
+{-| Converts a `Graph` into a valid DOT string.
+Note that you must supply conversion functions for node labels and edge labels to strings.
+-}
+output : (n -> Maybe String) -> (e -> Maybe String) -> Graph n e -> String
+output =
+    outputWithStyles defaultStyles
+
+
+{-| A type representing the attrs to apply at the graph, node, and edge entities (subgraphs and cluster subgraphs are not supported).
+
+Note that `Styles` is made up of strings, which loses type safety, but allows you to use any GraphViz attrs without having to model them out in entirety in this module. It is up to you to make sure you provide valid attr strings. See <http://www.graphviz.org/content/attrs> for available options.
+
+-}
+type alias Styles =
+    { rankdir : Rankdir
+    , graph : String
+    , node : String
+    , edge : String
+    }
+
+
+{-| Values to control the direction of the graph
+-}
+type Rankdir
+    = TB
+    | LR
+    | BT
+    | RL
+
+
+{-| A blank `Styles` record to build from to define your own styles.
+
+    myStyles =
+        { defaultStyles
+            | node = "shape=box, color=blue, style=\"rounded, filled\""
+        }
+
+-}
+defaultStyles : Styles
+defaultStyles =
+    Styles TB "" "" ""
+
+
+{-| Same as `output`, but allows you to add attrs to the graph.
+These attrs will be applied to the entire graph.
+-}
+outputWithStyles : Styles -> (n -> Maybe String) -> (e -> Maybe String) -> Graph n e -> String
+outputWithStyles styles mapNode mapEdge graph =
+    let
+        labelOnly maybeLabel =
+            case maybeLabel of
+                Nothing ->
+                    Dict.empty
+
+                Just l ->
+                    Dict.singleton "label" l
+    in
+    outputWithStylesAndAttributes styles (labelOnly << mapNode) (labelOnly << mapEdge) graph
+
+
+{-| Same as `outputWithStyles`, but allows each node and edge to include its
+own attrs. Note that you must supply a conversion function for node and edge
+labels that return a `Dict String String` of the attribute mappings.
+
+Note that you have to take care of setting the appropriate node and edge labels
+yourself.
+
+-}
+outputWithStylesAndAttributes :
+    Styles
+    -> (n -> Dict String String)
+    -> (e -> Dict String String)
+    -> Graph n e
+    -> String
+outputWithStylesAndAttributes styles nodeAttrs edgeAttrs graph =
+    let
+        attrAssocs : Dict String String -> String
+        attrAssocs =
+            Dict.toList
+                >> List.map (\( k, v ) -> k ++ "=\"" ++ v ++ "\"")
```

Comment:
I'd be a bit more careful with possible values of attribute `v`. If people insert `"` into attribute value (e.g. `this is my "great" label`) that might produce invalid attribute statement. How about using something like
```
\( k, v ) -> k ++ "=" ++ toString v
```
which would surround `v` with quotes AND escape whatever is in the String? That probably won't cover all the intricacies mentioned in the [DOT language](https://graphviz.gitlab.io/_pages/doc/info/lang.html) spec, but would be a bit more robust.